src/linker/Linker/Driver.cs

```diff
@@ -596,39 +604,25 @@ public bool Run (ILogger customLogger = null)
 
 		partial void PreProcessPipeline (Pipeline pipeline);
 
-		private bool ResolveCustomStepAssemblies (Stack<string> custom_steps, string[] search_directories, out List<Assembly> custom_steps_assemblies)
-		{
-			custom_steps_assemblies = new List<Assembly> ();
-			while (custom_steps.Peek ().EndsWith (".dll")) {
-				var assembly = custom_steps.Pop ();
-				if (Path.IsPathRooted (assembly)) {
-					var assemblyPath = Path.GetFullPath (assembly);
-					if (File.Exists (assemblyPath)) {
-						custom_steps_assemblies.Add (Assembly.LoadFrom (assemblyPath));
-						continue;
-					} else {
-						Console.WriteLine ($"Invalid assembly path '{assembly}' specified for '--custom-step' option");
-						return false;
-					}
-				} else {
-					bool assemblyFound = false;
-					foreach (var directory in search_directories) {
-						var assemblyPath = Path.Combine (directory, assembly);
-						if (File.Exists (assemblyPath)) {
-							custom_steps_assemblies.Add (Assembly.LoadFrom (assemblyPath));
-							assemblyFound = true;
-							break;
-						}
-					}
-
-					if (!assemblyFound) {
-						Console.WriteLine ($"The assembly '{assembly}' specified for '--custom-step' option could not be found");
-						return false;
-					}
+		protected static Assembly GetCustomAssembly (string arg, string[] search_directories) {
+			Assembly custom_assembly = null;
+			if (Path.IsPathRooted (arg)) {
+				var assemblyPath = Path.GetFullPath (arg);
+				if (File.Exists (assemblyPath))
+					return Assembly.LoadFrom (assemblyPath);
+				else
+					Console.WriteLine ($"Invalid assembly path '{arg}' specified for '--custom-step' option");
+			}
+			else {
+				foreach (var directory in search_directories) {
```

Comment:
I'm a bit nervous about this - mostly because we're combining two worlds of assemblies here. The `search_directories` are intended for resolution of assemblies for the app being linked. Now we're also using it to search for parts of the linker (the tool). This is problematic for several reasons:
* If there are potential ambiguous matches (the app and the linker using assembly with the same name) - not very likely, but still
* The linked app may not be of the same .NET Core version nor it may be the same target - for example I can link a linux app on windows. This means I'm searching through potentially linux specific directories when looking for windows assemblies.
* This also works the other way round - if I add a search directory specifically due to custom steps now I'm opening a potential issue that assemblies from that directly will be treated as part of the app.

For now I would prefer to only support the fully specified path and don't try to implement probing in any way. For the scenario we want this should not be a problem. It doesn't match the existing functionality on .NET Framework/Mono, but I don't think that's the goal we should be after.